EventFilter/L1TRawToDigi/src/implementations_stage2/IntermediateMuonUnpacker.cc

```diff
@@ -0,0 +1,122 @@
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "EventFilter/L1TRawToDigi/interface/Unpacker.h"
+
+#include "L1Trigger/L1TMuon/interface/MuonRawDigiTranslator.h"
+
+#include "GMTCollections.h"
+
+namespace l1t {
+   namespace stage2 {
+      class IntermediateMuonUnpacker : public Unpacker {
+         public:
+            virtual bool unpack(const Block& block, UnpackerCollections *coll) override;
+      };
+   }
+}
+
+// Implementation
+namespace l1t {
+   namespace stage2 {
+      bool
+      IntermediateMuonUnpacker::unpack(const Block& block, UnpackerCollections *coll)
+      {
+         LogDebug("L1T") << "Block ID  = " << block.header().getID() << " size = " << block.header().getSize();
+
+         auto payload = block.payload();
+
+         unsigned int nWords = 6; // every link transmits 6 words (3 muons) per bx
+         int nBX, firstBX, lastBX;
+         nBX = int(ceil(block.header().getSize() / nWords));
+         getBXRange(nBX, firstBX, lastBX);
+
+         // decide which collections to use according to the link ID
+         unsigned int linkId = (block.header().getID() - 1) / 2;
+         unsigned int coll1Cnt = 0;
+         MuonBxCollection* res1;
+         MuonBxCollection* res2;
+         switch (linkId) {
+           case 24:
```

Comment:
@rekovic you can use this commit https://github.com/thomreis/cmssw/commit/e7f409ceebb5d9e8a22a476464ff25282ee061be